### PR TITLE
Feature/recipe to grocery list

### LINKED ISSUE #16: recipe to grocery list functionality
part of the core functionality: add a button to recipe search that adds a recipe to grocery list, this should create a new grocery list if not already made, and add recipes there. a user should be able to view the grocery list either by recipes or by ingredients + quantities. grocery lists are user specific.

### PR DESCRIPTION
**Pull request by hhpfiona**
Recipe To Grocery List View implementation. We are still struggling to bring logged in user's data across views and to input data. however, in the interest of time, i think i will put this back until after phase 1.  code for view, controlle,r interactor etc is still available to be inspected.

### CODE REVIEW

File: src/main/java/entity/User.java [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
 import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
 
 public class User {
     private String userName;
     private String userEmail;
     private String userPassword;
     private LocalDateTime createdAt;
+    private List<Recipe> recipes;
```
Comment (parkhangorodsky) [inline src/main/java/entity/User.java:12]: what is recipes?

File: src/main/java/use_cases/_common/authentication/AuthenticationInterface.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ */
+public interface AuthenticationInterface {
+
+    boolean authenticate(String userEmail, String userPassword);
+
+    User getLoggedInUser();
+
+    void logout(User user);
+
+    void setLoggedInUser(User user); // Ensure this method is in the interface
```
Comment (parkhangorodsky) [inline src/main/java/use_cases/_common/authentication/AuthenticationInterface.java]: the comment here seems either not needed or misworded - won't the interface ensure the method is there?

File: src/main/resources/Examples.java
Comment (parkhangorodsky) [inline src/main/resources/Examples.java:1]: what is this file?

File: src/main/resources/example-recipe.json
Comment (parkhangorodsky) [inline src/main/resources/example-recipe.json:1]: do we need this? what is it?

File: target/classes/app/Config.class
Comment (parkhangorodsky) [inline target/classes/app/Config.class:1]: the target files don't need to put included in PRs

File: target/classes/entity/Ingredient.class
Comment (parkhangorodsky) [inline target/classes/entity/Ingredient.class:1]: the target files don't need to put included in PRs

File: target/classes/entity/Recipe.class
Comment (parkhangorodsky) [inline target/classes/entity/Recipe.class:1]: the target files don't need to put included in PRs

File: target/classes/entity/User.class
Comment (parkhangorodsky) [inline target/classes/entity/User.class:1]: the target files don't need to put included in PRs

**Review by parkhangorodsky — CHANGES_REQUESTED** [review]
overall looks good, some files feel like they don't need to be included - in the future probably best to only commit the relevant files.

left some comments

**Comment by parkhangorodsky** [comment]
link an issue to this PR

File: src/main/java/entity/User.java (hunk 1 above)
Comment (hhpfiona) [reply src/main/java/entity/User.java:12]: recipes associated with the users (recipes that the user has saved). planning to add methods such as bookmark or unbookmark recipes in teh future

File: src/main/java/use_cases/_common/authentication/AuthenticationInterface.java (hunk 2 above)
Comment (hhpfiona) [reply src/main/java/use_cases/_common/authentication/AuthenticationInterface.java]: i can take out that comment

File: src/main/resources/Examples.java
Comment (hhpfiona) [reply src/main/resources/Examples.java:1]: this should have gotten deleted - will do now

File: src/main/resources/example-recipe.json
Comment (hhpfiona) [reply src/main/resources/example-recipe.json:1]: ALSO should have been deleted, not sure why the change didn't go through. will delete now. sorry!

File: target/classes/app/Config.class
Comment (hhpfiona) [reply target/classes/app/Config.class:1]: i'm not sure why they're getting pushed actually, i'm sure i've set them to ignore. at the moment i'm not sure how to undo the push. any suggestions?

Comment (parkhangorodsky) [reply target/classes/app/Config.class:1]: not entirely sure why they show up in your changed files - but in the end you can just not commit them (even if they show up there).

File: src/main/java/entity/User.java (hunk 1 above)
Comment (parkhangorodsky) [reply src/main/java/entity/User.java:12]: sounds good - note for the future then: a better variable name might be good (like savedRecipes, or something like that)

**Review by parkhangorodsky — APPROVED** [review]

**Review by seonghyunban — APPROVED** [review]
Good!

**Pull request merged**